src/coreclr/src/debug/createdump/dumpwriter.cpp

```diff
@@ -171,24 +178,19 @@ DumpWriter::WriteDump()
     // Write memory region note headers
     for (const MemoryRegion& memoryRegion : m_crashInfo.MemoryRegions())
     {
-        phdr.p_flags = memoryRegion.Permissions();
-        phdr.p_vaddr = memoryRegion.StartAddress();
-        phdr.p_memsz = memoryRegion.Size();
-
         if (memoryRegion.IsBackedByMemory())
```

Comment:
I thought reading the symbols to find the DAC required us to save some Dwarf sections that weren't backed by memory.  Perhaps they are not here.

Reply:
Not anymore even on Linux.  It is worse for lldb not to fail (returning zero for this sections) causing me problems/hacks when detecting the metadata in a PE module is in-memory or not.

Reply:
Does it mean we need the file for the Cross DAC to work now?  i.e. Will it no longer be in the core dump?  Or is it added to the dump a different way?

While were at it, will the coredumps created by Linux w/o createdump have the sections we need?

Reply:
This code is adding all the memory regions not the module info if that is what you mean about "file".  The NT_FILE records have the list of modules (including the PE assemblies because they are memory mapped by the runtime). See the DumpWriter::WriteNTFileInfo function.

"backed by memory" basically means that is readable by createdump.  

If you set the coredump_filter to 0x3f because running the app, Linux system dumps should have all the state needed and they only have the backed by memory regions. The NT_FILE also have everything we need.